app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -295,6 +333,14 @@ def ca_cert
 
     private
 
+    def os_name_mapping(host)
+      host.operatingsystem.name =~ /redhat|centos/i ? 'rhel': host.operatingsystem.name
+    end
+
+    def arch_name_mapping(host)
+      host.architecture.name == 'x86_64' ? 'x64' : host.architecture.name if host.architecture
```

Comment:
Because the operating system is mandatory to be able even detect the os to be used. Therefore we check that in the `determine_os_type` right at the beginning, while the architecture is just a hint, but we can still determine os without it (there is `rhel_7x86` and `rhel_7` type in ovrit)